src/backend/providers/playnite/PlayniteProvider.cpp

```diff
@@ -0,0 +1,306 @@
+            game.developerList().append(developer_it->second);
+        }
+    }
+
+    for (const QString& publisher_id : game_info.publisher_ids) {
+        auto publisher_it = components.companies.find(publisher_id);
+        if (publisher_it != components.companies.end()) {
+            game.publisherList().append(publisher_it->second);
+        }
+    }
+
+    for (const QString& genre_id : game_info.genre_ids) {
+        auto genre_it = components.genres.find(genre_id);
+        if (genre_it != components.genres.end()) {
+            game.genreList().append(genre_it->second);
+        }
+    }
+
+    if (!game.releaseDate().isValid())
+        game.setReleaseDate(QDate::fromString(game_info.release_date, Qt::ISODate));
+
+    if (game_info.community_score >= 0.0f) {
+        game.setRating(game_info.community_score / 100.0f);
+    }
+
+    const QString playnite_dir_path = playnite_dir.absolutePath();
+    if (!game_info.cover_image.isEmpty()) {
+        const QString cover_path = QStringLiteral("%1/library/files/%2").arg(playnite_dir_path, game_info.cover_image);
+        game.assetsMut().add_file(AssetType::BOX_FRONT, cover_path);
+    }
+
+    if (!game_info.background_image.isEmpty()) {
+        const QString bg_path =
+            QStringLiteral("%1/library/files/%2").arg(playnite_dir_path, game_info.background_image);
+        game.assetsMut().add_file(AssetType::BACKGROUND, bg_path);
+    }
+}
+} // namespace
+
+PlayniteProvider::PlayniteProvider(QObject* parent)
+    : Provider(QLatin1String("playnite"), QStringLiteral("Playnite"), parent)
+{}
+
+Provider& PlayniteProvider::run(SearchContext& sctx)
+{
+    const auto playnite_dir_path = [this] {
+        const auto option_it = options().find(QStringLiteral("installdir"));
+        return (option_it != options().cend()) ? QDir::cleanPath(option_it->second.front()) + QLatin1Char('/')
+                                               : default_installation();
+    }();
+
+    if (playnite_dir_path.isEmpty() || !QFileInfo::exists(playnite_dir_path)) {
+        Log::info(display_name(), LOGMSG("No installation found"));
+        return *this;
+    }
+
+    Log::info(
+        display_name(), LOGMSG("Looking for installation at `%1`").arg(QDir::toNativeSeparators(playnite_dir_path)));
+
+    const std::vector<model::Game*> games = create_games(playnite_dir_path, sctx);
+
+    if (games.empty()) {
+        Log::warning(display_name(), LOGMSG("No games found"));
+    }
+    return *this;
+}
+
+std::vector<model::Game*> PlayniteProvider::create_games(const QString& playnite_path, SearchContext& sctx)
+{
+    const QDir playnite_dir(playnite_path);
+    const PlayniteMetadataParser playnite_metadata_parser(display_name(), playnite_path);
+    const PlayniteComponents& components = playnite_metadata_parser.parse_metdata();
+
+    std::vector<model::Game*> output;
+    PlayniteLaunchHelper launch_helper;
+
+    const float num_progress_updates = 5;
```

Comment:
Is there a particular reason why this is exactly 5?

Reply:
Just a number I made up that seemed like "not too many progress updates".